### ISSUE TITLE
add minShares to vault withdrawals

### ISSUE THREAD

**Issue by lbertenasco** [yearn/yearn-vaults#477]
it's not the best UX when you desire to withdraw 100 shares that it will only consume 50.
should add `minShares` as a parameter and revert if it cannot fill the withdraw

**Comment by pandadefi** [comment]
You end up spending a lot of gas and at the end it will revert, does that make sense?

**Comment by ankitiitb1069** [comment]
Added changes here https://github.com/ankitiitb1069/yearn-vaults/tree/add-minshares-to-vault-withdrawal, can I raise a PR for this now?

**Comment by pandadefi** [comment]
You must add tests to your changes

Le mar. 2 nov. 2021, 21:28, Ankit Parashar ***@***.***> a
écrit :

**Comment by lbertenasco** [comment]
> You end up spending a lot of gas and at the end it will revert, does that make sense?

yes, because metamask will tell you it's a reverted tx, and you won't end up sending that tx at all.

also @ankitiitb1069 please add tests and submit a PR and we'll gladly review it :)

**Comment by parseb** [comment]
@ankitiitb1069 did you submit a PR for this?

**Comment by pandadefi** [comment]
@parseb he didn't

**Comment by parseb** [comment]
@pandadefi on it, if PR not linked in 48h, assume the worst.

**Comment by parseb** [comment]
Spent hours trying to fix this. 
Not the only one getting this error.
I'm also likely the worst person to try and solve it since I'm new to everything in this env.

![image](https://user-images.githubusercontent.com/5999852/150334276-2ae650a6-1504-4e9c-bd27-f407acdd3ec7.png)

**Comment by gmridul** [comment]
Hey @parseb, i'll try to give this a shot if you're okay?

**Comment by gmridul** [comment]
@pandadefi, sent this PR #517.

**Comment by pandadefi** [comment]
We are working on a new version of the vault that will not be based on this code. We aren't going to make a new release on top of that.

**Comment by gmridul** [comment]
got it, I'll close the PR.

**Comment by gmridul** [comment]
@pandadefi  any help you need on the new version of vaults?

**Comment by pandadefi** [comment]
We are still in the planning phase
